### PR TITLE
modules: openthread: adjust to unified OT TX/RX timestamp definition

### PR DESCRIPTION
**Pull request by ghost**
This change prepares for the time when OT may introduce a common RX/TX timestamp definition based on the standard's SFD. If they decide to keep their current definition, it can simply be discarded.
    
This change needs to be synchronized with the upstream Thread implementation. An appropriate change set has been prepared for OpenThread, see https://github.com/openthread/openthread/pull/9322 (upstream) and https://github.com/zephyrproject-rtos/openthread/pull/113 (zephyr openthread module).
    
~TODO: The commit and PR are marked DNM because west.yml still needs to be updated with the correct upstream commit SHA for OpenThread once the module has been updated.~

UPDATE: Done.

### CODE REVIEW

**Comment by fgrandel** [comment]
Closed due to upstream inactivity.

**Comment by edmont** [comment]
Hi @fgrandel, OT PR has been finally merged. Would you mind reopening this one?

**Comment by fgrandel** [comment]
> Hi @fgrandel, OT PR has been finally merged. Would you mind reopening this one?

Thanks @edmont for the good news. Of course I reopen. :-)

**Comment by edmont** [comment]
Could you please extend https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/openthread/CMakeLists.txt with newly added options?
- `OT_DEVICE_PROP_LEADER_WEIGHT`
- `OT_LINK_METRICS_MANAGER`

**Comment by fgrandel** [comment]
> Could you please extend `modules/openthread/CMakeLists.txt` and `modules/openthread/Kconfig.features` with newly added options?
> 
> * `OT_DEVICE_PROP_LEADER_WEIGHT`
> * `OT_LINK_METRICS_MANAGER`

@edmont  Yes sure, you'd have to lend me a hand, though, as I have no idea what those are about as they're unrelated to my own changes. Should we quickly co-ordinate over Discord?

**Comment by edmont** [comment]
> @edmont Yes sure, you'd have to lend me a hand, though, as I have no idea what those are about as they're unrelated to my own changes. Should we quickly co-ordinate over Discord?

It's just a matter of making the options available through Kconfig. We use to do that every time new OT is bumped not to forget about them later on. Take a look to the previous OT upmerge changes as an example: https://github.com/zephyrproject-rtos/zephyr/pull/60379/files

File: modules/openthread/Kconfig.features [hunk 1]
```diff
@@ -92,6 +92,12 @@ config OPENTHREAD_CSL_RECEIVER
 	help
 	  Enable CSL Receiver support for Thread 1.2
 
+config OPENTHREAD_DEVICE_PROP_LEADER_WEIGHT
+	bool "Enable device props for leader weight"
```
Comment (edmont) [inline modules/openthread/Kconfig.features:96]: See compliance check error:

> Boolean option 'OPENTHREAD_DEVICE_PROP_LEADER_WEIGHT' prompt must not start with 'Enable...'. Please check Kconfig guidelines.

**Review by edmont — APPROVED** [review]

File: modules/openthread/Kconfig.features (hunk 1 above)
Comment (fgrandel) [reply modules/openthread/Kconfig.features]: Fixed.

File: modules/openthread/Kconfig.features [hunk 2]
```diff
@@ -92,6 +92,12 @@ config OPENTHREAD_CSL_RECEIVER
 	help
 	  Enable CSL Receiver support for Thread 1.2
 
+config OPENTHREAD_DEVICE_PROP_LEADER_WEIGHT
+	bool "Device props for leader weight"
+	help
+	  Enable the device properties which are then used to determine and set
+	  the Leader Weight.
+
```
Comment (maciejbaczmanski) [inline modules/openthread/Kconfig.features:100]: I'm wondering if we should add some dependency with default regarding OT version here, as in OT, the flag is enabled by default for version 1.3.1 and later: https://github.com/nrfconnect/sdk-openthread/blob/79a3a26db601571829d803b61d6b390a87ea9810/src/core/config/mle.h#L101
Thoughts?

Comment (fgrandel) [reply modules/openthread/Kconfig.features:100]: @maciejbaczmanski @edmont Is there a way to do a `default y if THREAD_V131` or similar? I seem to remember we had a variable like this, just don't remember the exact syntax.

Comment (fgrandel) [reply modules/openthread/Kconfig.features]: @maciejbaczmanski @edmont Fixed (just inversed the logic to avoid the "not" operation) - is this ok?

**Review by maciejbaczmanski — APPROVED** [review]
LGTM, Thanks!

**Review by edmont — APPROVED** [review]

**Comment by edmont** [comment]
@fgrandel show `[DNM] ` be removed from the PR title?

**Comment by fgrandel** [comment]
> @fgrandel show `[DNM] ` be removed from the PR title?

As long as https://github.com/zephyrproject-rtos/openthread/pull/113 is not merged this must necessarily be DNM. I need to switch from the pull request to the SHA in this branch before I can remove the DNM marker (and label).

**Review by canisLupus1313 — DISMISSED** [review]

File: west.yml [hunk 3]
```diff
@@ -303,7 +303,7 @@ manifest:
       revision: 42b7c577714b8f22ce82a901e19c1814af4609a8
       path: modules/lib/open-amp
     - name: openthread
-      revision: f7690fe7e9d638341921808cba6a3e695ec0131e
+      revision: pull/113/head
```
Comment (rlubos) [inline west.yml:306]: Thanks, I've updated the OT fork, please update SHA: `d62167ee34b091e7025c9ec2820aae71e17a3944` (please remember to remove DNM from the commit title).

Comment (fgrandel) [reply west.yml]: Both: done. :-) Thanks for reviewing/merging!

**Review by pdgendt — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by canisLupus1313 — APPROVED** [review]

**Pull request merged**